File: backend/app/models/agent.py
"""
Agent model.
"""
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class Agent(BaseModel):
    """Agent model representing an AI agent."""

    id: Optional[str] = Field(None, alias="_id")
    name: str
    description: str
    type: str  # "assistant", "researcher", "coder", etc.
    status: str = "inactive"  # "active", "inactive", "error"
    model: str = "gpt-4"
    system_prompt: Optional[str] = None
    tools: List[str] = []
    capabilities: List[str] = []
    config: Dict[str, Any] = {}
    platform_id: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: Optional[datetime] = None

    class Config:
        populate_by_name = True

File: frontend/src/types/agent.ts
export type AgentStatus = 'active' | 'inactive' | 'error' | 'initializing';

export interface AgentConfig {
  model: string;
  temperature?: number;
  maxTokens?: number;
  systemPrompt?: string;
  timeout?: number;
  retryAttempts?: number;
}

export interface Agent {
  id: string;
  name: string;
  description: string;
  status: AgentStatus;
  config: AgentConfig;
  tools: string[];
  createdAt: string;
  updatedAt: string;
  lastExecutedAt?: string;
  executionCount: number;
  successRate: number;
}

export interface CreateAgentRequest {
  name: string;
  description: string;
  config: AgentConfig;
  tools?: string[];
}

export interface UpdateAgentRequest {
  name?: string;
  description?: string;
  config?: Partial<AgentConfig>;
  tools?: string[];
  status?: AgentStatus;
}

export interface AgentExecution {
  id: string;
  agentId: string;
  status: 'running' | 'completed' | 'failed' | 'cancelled';
  input: Record<string, unknown>;
  output?: Record<string, unknown>;
  error?: string;
  startedAt: string;
  completedAt?: string;
  duration?: number;
}
